src/libsystemd/sd-device/sd-device.c

```diff
@@ -325,6 +325,10 @@ _public_ int sd_device_new_from_subsystem_sysname(sd_device **ret, const char *s
         if (access(syspath, F_OK) >= 0)
                 return sd_device_new_from_syspath(ret, syspath);
 
+        syspath = strjoina("/sys/firmware/", subsystem, "/", sysname);
```

Comment:
@whot, PTAL.

Using `sysname` not `name` is a bit surprising, but I see that you want to get at `devicetree/base`, so OK.